### ISSUE TITLE
Project: Source Maps Wizard

### ISSUE THREAD

**Issue by Lms24** [getsentry/team-sdks#16]
This Epic tracks all the work related to creating Wizard flows for uploading source maps with various tools or bundlers.

First, we should start off with some preparations to our currently existing wizard flows
```[tasklist]
### Preparations
- [ ] https://github.com/getsentry/sentry-wizard/issues/292
- [ ] https://github.com/getsentry/sentry-wizard/issues/293
- [ ] https://github.com/getsentry/sentry-wizard/issues/296
- [ ] https://github.com/getsentry/sentry-wizard/issues/301
- [ ] https://github.com/getsentry/sentry-javascript-bundler-plugins/issues/332
```
**Goal**: We can reuse our login flow across all (JS) wizards and we have a base to start with the source map flows

---

This is what we'll ship as a first version:
```[tasklist]
### First Iteration
- [ ] https://github.com/getsentry/sentry-wizard/issues/298
- [ ] https://github.com/getsentry/sentry-wizard/issues/303
- [ ] https://github.com/getsentry/sentry-wizard/issues/305
- [ ] https://github.com/getsentry/sentry-wizard/issues/312
- [ ] https://github.com/getsentry/sentry-docs/issues/7300
```
**Goal**: Have a solid overall flow in place for the most important tools but we'll omit the more "magical" aspects (auto-detection, AST modification) for future iterations. Users will be asked to copy tailored code snippets for adding plugins to their bundler configs.

---

Following our first iteration, we'll work refining and improving the first iteration with fixes, small features and a little more auto-detection: 

```[tasklist]
### Improvements after first iteration
- [ ] https://github.com/getsentry/sentry-wizard/pull/328
- [ ] https://github.com/getsentry/sentry-wizard/pull/331
- [ ] https://github.com/getsentry/sentry-wizard/pull/335
- [ ] https://github.com/getsentry/sentry-wizard/pull/336
- [ ] https://github.com/getsentry/sentry-wizard/pull/338
- [ ] https://github.com/getsentry/sentry-wizard/pull/341
- [ ] https://github.com/getsentry/sentry-wizard/pull/344
- [ ] https://github.com/getsentry/sentry-wizard/pull/349
- [ ] https://github.com/getsentry/sentry-wizard/pull/354
- [ ] https://github.com/getsentry/sentry-wizard/pull/359
- [ ] https://github.com/getsentry/sentry-wizard/pull/360
- [ ] https://github.com/getsentry/sentry-wizard/pull/368
- [ ] https://github.com/getsentry/sentry-wizard/issues/364
- [ ] https://github.com/getsentry/sentry-wizard/pull/369
- [ ] https://github.com/getsentry/sentry-wizard/pull/372
```

 **Goal**: A more friction-less flow that accounts for missing pieces in the user setup, detects better options, supports special platforms and generally makes the wizard more usable.

---

Once we completed improving and refining our initial version, let's add some more 🪄 

```[tasklist]
### Advanced Features
- [ ] https://github.com/getsentry/sentry-wizard/issues/366
- [ ] https://github.com/getsentry/sentry-wizard/issues/384
- [ ] https://github.com/getsentry/sentry-wizard/issues/394
```

 **Goal**: A more friction-less flow that accounts for missing pieces in the user setup, detects better options, supports special platforms and generally makes the wizard more usable.

---

### Stretch Goals & Miscellaneous
* Think about CI configs
* Provide instructions for UglifyJS
* Provide instructions for SystemJS
* Check for correct SDK (e.g. user uses react in NextJS or @sentry/angular in NG12+) and give a hint that there might be a better SDK available. This is not directly related to source maps though.
* Special case Electron-forge?
* More AST for other bundlers. Webpack with highest prio 
* Add `--org`, `--project`, `--authToken` options
* [Waiting on contractor] Figure out capacitor, ionic

ref:
getsentry/sentry-wizard#289 
getsentry/sentry-wizard#288 
https://github.com/getsentry/team-webplatform-meta/issues/71

**Comment by Lms24** [comment]
Closing this project issue as our initiative around introducing the source maps wizard was completed with the release of version [3.10.0]() of the Sentry wizard. 

We will make further improvements down the line and track them as separate issues but I think at this point, it's fair to close the initial initiative.
